### PR TITLE
Fix compatible for webhook ref type

### LINKED ISSUE #25185: Webhook create events return incorrect `ref_type`
### Description

Since commit f9cfd6ce5bd7f27e2655fd307022461a802fc49c (PR #24634), Gitea webhooks for `create` events have a `ref_type` of `tags`, rather than `tag` as it used to be and currently is on GitHub. This breaks webhook receivers such as Woodpecker CI. I don't think this change was intentional. If it was, I'll file an issue on the Woodpecker repo instead.

### Gitea Version

1.20.0+rc0-9-gde1d14590

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

2.40.1

### Operating System

Debian Bullseye

### How are you running Gitea?

Official docker container, tag `1.20-nightly`

### Database

SQLite

### PR DESCRIPTION
**Pull request by lunny**
Fix #25185 
Caused by #24634

### CODE REVIEW

**Comment by wxiaoguang** [comment]
Incomplete. There are others.

image

image

**Comment by lunny** [comment]
@wxiaoguang It's ready now.

File: services/webhook/notifier.go [hunk 1]
```diff
@@ -752,10 +752,20 @@ func (m *webhookNotifier) NotifyCreateRef(ctx context.Context, pusher *user_mode
...
+	var refType string
+	if refFullName.IsBranch() {
+		refType = "branch"
+	} else if refFullName.IsTag() {
+		refType = "tag"
+	} else {
+		log.Error("unsupported ref type for NotifyCreateRef: %s", refFullName)
+		return
+	}
+
```
Comment (wxiaoguang) [inline services/webhook/notifier.go]: Duplicate?

Comment (lunny) [reply services/webhook/notifier.go]: Done.

**Review by wxiaoguang — APPROVED** [review]

**Review by wolfogre — APPROVED** [review]

**Pull request merged**